### PR TITLE
Break reference cycle

### PR DESCRIPTION
**Pull request by sigurdm**
Depends on https://github.com/flutter/engine/pull/4354 to avoid a double release.

### CODE REVIEW

**Comment by sigurdm** [comment]
Reuploaded without the formatting noise and logging.

File: ios/Classes/VideoPlayerPlugin.m [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+@end
+
+@interface VideoPlayer : NSObject<FlutterTexture>
 @property(readonly, nonatomic) AVPlayer* player;
 @property(readonly, nonatomic) AVPlayerItemVideoOutput* videoOutput;
 @property(readonly, nonatomic) CADisplayLink* displayLink;
 @property(nonatomic, copy) void (^onFrameAvailable)();
-- (instancetype)initWithURL:(NSURL*)url;
+- (instancetype)initWithURL:(NSURL*)url
+           withFrameUpdater:(FrameUpdater*)frameUpdater;
```
Comment (mravn-google) [inline ios/Classes/VideoPlayerPlugin.m]: I'd shorten that to `frameUpdater:`

File: ios/Classes/VideoPlayerPlugin.m [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
-@property(readonly, nonatomic) int64_t textureId;
+@interface FrameUpdater : NSObject
+@property(nonatomic) int64_t textureId;
+@property(nonatomic, readonly) NSObject<FlutterTextureRegistry>* registry;
+- (void)onDisplayLink:(CADisplayLink*)link;
+@end
+
+@implementation FrameUpdater
+- (FrameUpdater*)initWithRegistry:(NSObject<FlutterTextureRegistry>*)registry {
+  self = [super init];
```
Comment (mravn-google) [inline ios/Classes/VideoPlayerPlugin.m]: Check that `self` is not `nil`.

File: ios/Classes/VideoPlayerPlugin.m [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+- (void)onDisplayLink:(CADisplayLink*)link {
+  [_registry textureFrameAvailable:_textureId];
+}
+@end
+
+@interface VideoPlayer : NSObject<FlutterTexture>
 @property(readonly, nonatomic) AVPlayer* player;
 @property(readonly, nonatomic) AVPlayerItemVideoOutput* videoOutput;
 @property(readonly, nonatomic) CADisplayLink* displayLink;
 @property(nonatomic, copy) void (^onFrameAvailable)();
```
Comment (mravn-google) [inline ios/Classes/VideoPlayerPlugin.m]: Remove this?

File: ios/Classes/VideoPlayerPlugin.m (hunk 2 above)
Comment (sigurdm) [reply ios/Classes/VideoPlayerPlugin.m]: done

File: ios/Classes/VideoPlayerPlugin.m (hunk 1 above)
Comment (sigurdm) [reply ios/Classes/VideoPlayerPlugin.m]: done

File: ios/Classes/VideoPlayerPlugin.m (hunk 3 above)
Comment (sigurdm) [reply ios/Classes/VideoPlayerPlugin.m]: done

**Pull request merged**